init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/config/GitRepo.java

```diff
@@ -0,0 +1,100 @@
+package uk.ac.ox.ndph.mts.init_service.config;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.JGitInternalException;
+import org.eclipse.jgit.lib.Constants;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.ObjectLoader;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevTree;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.treewalk.TreeWalk;
+import org.eclipse.jgit.treewalk.filter.PathFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import uk.ac.ox.ndph.mts.init_service.exception.InvalidConfigException;
+
+import javax.annotation.PostConstruct;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+@Component
+public class GitRepo {
```

Comment:
I thought the plan is to get the configuration from the config-server eventually. If so, doing this with git now can be avoided with just fetching the file from local resources. We can skip this functionality since it will away be replaced real soon...

Reply:
In this first iteration before we integrate the config-server i'd like this to remain. 

QA whilst testing will be able to add a new config file to the repo https://github.com/NDPH-ARTS/global-trial-config

QA can then modify docker-compose to  

environment:
        CONFIG_JSON: config-init.json 

spin up and then test.